BuildResidentialHPXML/measure.rb

```diff
@@ -3567,33 +3610,56 @@ def self.set_header(hpxml, args)
         natvent_availability = args[:schedules_power_outage_window_natvent_availability].get
       end
 
-      hpxml.header.unavailable_periods.add(column_name: 'Power Outage', begin_month: begin_month, begin_day: begin_day, begin_hour: begin_hour, end_month: end_month, end_day: end_day, end_hour: end_hour, natvent_availability: natvent_availability)
+      if not unavailable_period_exists(hpxml, 'Power Outage', begin_month, begin_day, begin_hour, end_month, end_day, end_hour, natvent_availability)
+        hpxml.header.unavailable_periods.add(column_name: 'Power Outage', begin_month: begin_month, begin_day: begin_day, begin_hour: begin_hour, end_month: end_month, end_day: end_day, end_hour: end_hour, natvent_availability: natvent_availability)
+      end
     end
 
     if args[:software_info_program_used].is_initialized
+      if !hpxml.header.software_program_used.nil? && (hpxml.header.software_program_used != args[:software_info_program_used].get)
+        errors << "hpxml.header.software_program_used=#{hpxml.header.software_program_used}; cannot set software_info_program_used=#{args[:software_info_program_used].get}."
```

Comment:
These error messages are so cryptic, a user is not going to understand what they mean or why they are getting them. Please use something more easily understandable, like simply "'Software Info: Program Used' cannot vary across dwelling units."

Reply:
Or you could use similar language as what's used in the HPXMLtoOS measure:
https://github.com/NREL/OpenStudio-HPXML/blob/b63b0b8e6a624747fd2136de66f9b11a7a0e55e6/HPXMLtoOpenStudio/measure.rb#L258